### PR TITLE
Remove best ms2 datatype

### PR DESCRIPTION
**Pull request by robinschmid**
Its bad to separate the AllFragmentSpectraType and BestFragmentSpectraType.

Therefore, I removed the best type. Now we only use a sorted list of fragment scans - first element is the best scan

### CODE REVIEW

**Review by SteffenHeu — COMMENTED** [review]
Some comments, please tell me what you think

File: src/main/java/io/github/mzmine/datamodel/features/ModularFeature.java [hunk 1]
```diff
@@ -111,21 +111,20 @@ public ModularFeature(@NotNull ModularFeatureList flist) {
...
+      @NotNull Range<Float> intensityRange) {
     this(flist, dataFile, mz, rt, height, area, Arrays.asList(scanNumbers),
         DataPointUtils.getMZsAsDoubleArray(dataPointsPerScan),
         DataPointUtils.getIntenstiesAsDoubleArray(dataPointsPerScan), featureStatus,
-        representativeScan, fragmentScanNumber, allMS2FragmentScanNumbers, rtRange, mzRange,
-        intensityRange);
+        representativeScan, allMS2FragmentScanNumbers, rtRange, mzRange, intensityRange);
   }
 
   public ModularFeature(ModularFeatureList flist, RawDataFile dataFile, double mz, float rt,
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/features/ModularFeature.java:123]: true

File: src/main/java/io/github/mzmine/datamodel/features/Feature.java [hunk 2]
```diff
@@ -160,35 +156,35 @@ default float getRetentionTimeAtIndex(int i) {
...
 
   /**
-   * Set all fragment scan numbers
+   * Set all fragmentation scans. First element is "best" representative scan
    *
-   * @param allMS2FragmentScanNumbers
+   * @param allMS2FragmentScanNumbers usually sorted by most intense scans first (represantative
+   *                                  scan as first element)
    */
   void setAllMS2FragmentScans(List<Scan> allMS2FragmentScanNumbers);
```
[the remark this answers is not in the document]
Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/datamodel/features/Feature.java:189]: Or that. 
i was thinking about addFragmentScan(Scan scan, boolean sort); but can also do addFragmentScanAndSort(Scan... scan); (correct order)

File: src/main/java/io/github/mzmine/datamodel/features/ModularFeature.java [hunk 3]
```diff
@@ -355,12 +352,24 @@ public DataPoint getDataPoint(Scan scan) {
...
+    List<Scan> allMS2 = get(FragmentScanNumbersType.class);
+    if (allMS2 != null && !allMS2.isEmpty()) {
+      return allMS2.get(0);
+    } else {
+      return null;
+    }
   }
 
   @Override
   public void setFragmentScan(Scan fragmentScan) {
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/features/ModularFeature.java:364]: was unused anyway so I removed it.
We can have a method when its actually used

File: src/main/java/io/github/mzmine/datamodel/features/Feature.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/features/Feature.java:189]: I removed the addFragmentScan method completely as it is not used anymore. And added a method to set and sort a list of fragment scans

**Comment by robinschmid** [comment]
Ready

**Comment by SteffenHeu** [comment]
Some tests are failing

**Comment by SteffenHeu** [comment]
Looks like a merge conflictr due to the multi charge pattern

**Comment by robinschmid** [comment]
merged with @SteffenHeu

**Pull request merged**